Sources/NIOCore/Utilities.swift

```diff
@@ -22,3 +22,11 @@ internal func debugOnly(_ body: () -> Void) {
     // FIXME: duplicated with NIO.
     assert({ body(); return true }())
 }
+
+/// Allows to "box" another value.
+final class Box<T> {
+    // FIXME: Duplicated with NIO.
```

Comment:
Can't we make this public so that NIO gets it too, and avoid the duplication?

Reply:
Or use spi if we don't want it to be public?

Reply:
Yes, in principle we can. I'd like to defer that API design question to a later PR for now, as this type is exceedingly trivial, but we do have a number of `FIXME`s around duplicated code that needs to be addressed. I think it should be case-by-case though, as some things will benefit from just being separate (e.g. `Box` is the kind of thing that we maybe just want to keep entirely separate).